common/src/main/java/com/sequenceiq/cloudbreak/util/PasswordUtil.java

```diff
@@ -1,28 +1,31 @@
 package com.sequenceiq.cloudbreak.util;
 
-import java.math.BigInteger;
-import java.security.MessageDigest;
-import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
 
 import org.apache.commons.lang3.RandomStringUtils;
 
 public class PasswordUtil {
 
-    private static final int PWD_LENGTH = 128;
+    private static final int PWD_LENGTH = 26;
 
-    private static final int RADIX = 32;
+    private static final SecureRandom SECURE_RANDOM = new SecureRandom();
 
     private PasswordUtil() {
     }
 
     public static String generatePassword() {
-        try {
-            MessageDigest messageDigest = MessageDigest.getInstance("MD5");
-            String raw = RandomStringUtils.randomAscii(PWD_LENGTH);
-            byte[] digest = messageDigest.digest(raw.getBytes());
-            return new BigInteger(1, digest).toString(RADIX);
-        } catch (NoSuchAlgorithmException e) {
-            throw new RuntimeException(e);
-        }
+        return getRandomLettersAndNumbersWithCount(PWD_LENGTH, SECURE_RANDOM);
+    }
+
+    public static String getRandomLettersAndNumbersWithCount(int count, SecureRandom secureRandom) {
```

Comment:
This method doesn't need `secureRandom` as an argument, please remove that. Also the `WithCount` can be removed since other methods consume count as an argument as well without the `WithCount` in their names.